packages/hash/api/src/graphql/resolvers/knowledge/page/update-page.ts

```diff
@@ -0,0 +1,40 @@
+import { WORKSPACE_TYPES } from "../../../../graph/workspace-types";
+import { PageModel } from "../../../../model";
+import {
+  MutationUpdateKnowledgePageArgs,
+  ResolverFn,
+} from "../../../apiTypes.gen";
+import { LoggedInGraphQLContext } from "../../../context";
+import {
+  UnresolvedKnowledgePageGQL,
+  mapPageModelToGQL,
+} from "../model-mapping";
+
+export const updateKnowledgePage: ResolverFn<
+  Promise<UnresolvedKnowledgePageGQL>,
+  {},
+  LoggedInGraphQLContext,
+  MutationUpdateKnowledgePageArgs
+> = async (
+  _,
+  { entityId, updatedProperties },
+  { dataSources: { graphApi } },
+) => {
+  const page = await PageModel.getPageById(graphApi, { entityId });
+
+  const updatedPageEntity = await page.updateProperties(graphApi, {
+    updatedProperties: Object.entries(updatedProperties).map(
+      ([propertyName, value]) => ({
+        propertyTypeBaseUri:
+          WORKSPACE_TYPES.propertyType[
+            propertyName as keyof MutationUpdateKnowledgePageArgs["updatedProperties"]
```

Comment:
This puts an implicit "binding" between the GQL types and the names of the properties in the backend, do we expect we have to do this more? (probably have to do it for all `updateKnowledge*` mutation that allows properties). I don't think we need to do it in this PR, but this might be useful to wrap in a helper

Reply:
> probably have to do it for all updateKnowledge*

We don't have any other `updateKnolwedge*` currently where this can be used, and I'm not sure what a helper would look like for future ones because it will depend on the base type for the arguments? Could you give an example of what re-usable code could look like? Maybe I'm not understanding

Reply:
I anticipate that we'll have something similar for Comments (validating text tokens) and for other workspace types that have properties that are set through GQL types. I don't think we should add a helper function to handle these  just yet, though. Just flagging that we'll see this type of pattern